code/modules/ninja/energy_katana.dm

```diff
@@ -40,13 +39,12 @@
 	spark_system.set_up(5, 0, src)
 	spark_system.attach(src)
 
-/obj/item/energy_katana/attack_self(mob/user)
-	dash_toggled = !dash_toggled
-	to_chat(user, "<span class='notice'>You [dash_toggled ? "enable" : "disable"] the dash function on [src].</span>")
-
 /obj/item/energy_katana/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
-	if(dash_toggled && !Adjacent(target) && !target.density)
+
+	var/list/params = params2list(click_parameters)
+
+	if(params["right"] && !Adjacent(target) && !target.density)
```

Comment:
now that this is its own button, can we remove the adjacency check so that you don't have to back up a tile to teleport into something adjacent to you?